lib/rucio/core/did_meta_plugins/json_meta.py

```diff
@@ -150,41 +139,77 @@ def delete_metadata(self, scope, name, key, session=None):
-
-        if long:
-            for row in query.yield_per(5):
-                yield {
-                    'scope': row.scope,
-                    'name': row.name,
-                    'did_type': 'Info not available in JSON Plugin',
-                    'bytes': 'Info not available in JSON Plugin',
-                    'length': 'Info not available in JSON Plugin'
-                }
-        else:
-            for row in query.yield_per(5):
-                yield row.name
+        if not ignore_dids:
+            ignore_dids = set()
+
+        # backwards compatability for filters as single {}.
+        if isinstance(filters, dict):
+            filters = [filters]
+
+        # instantiate fe and create sqla query, note that coercion to a model keyword
+        # is not appropriate here as the filter words are stored in a single json column.
+        fe = FilterEngine(filters, model_class=models.DidMeta, strict_coerce=False)
+        query = fe.create_sqla_query(
+            additional_model_attributes=[
+                models.DidMeta.scope,
+                models.DidMeta.name
+            ], additional_filters=[
+                (models.DidMeta.scope, operator.eq, scope)
+            ],
+            json_column=models.DidMeta.meta
+        )
+
+        if limit:
+            query = query.limit(limit)
+        if recursive:
+            from rucio.core.did import list_content
+
+            # Get attached DIDs and save in list because query has to be finished before starting a new one in the recursion
+            collections_content = []
+            for did in query.yield_per(100):
+                if (did.did_type == DIDType.CONTAINER or did.did_type == DIDType.DATASET):
+                    collections_content += [d for d in list_content(scope=did.scope, name=did.name)]
+
+            # Replace any name filtering with recursed DID names.
+            for did in collections_content:
+                for or_group in filters:
+                    or_group['name'] = did['name']
+                for result in self.list_dids(scope=did['scope'], filters=filters, recursive=True, did_type=did_type, limit=limit, offset=offset,
+                                             long=long, ignore_dids=ignore_dids, session=session):
+                    yield result
+
+        try:
+            for did in query.yield_per(5):                  # don't unpack this as it makes it dependent on query return order!
+                if long:
+                    did_full = "{}:{}".format(did.scope, did.name)
+                    if did_full not in ignore_dids:         # concatenating results of OR clauses may contain duplicate DIDs if query result sets not mutually exclusive.
+                        ignore_dids.add(did_full)
+                        yield {
+                            'scope': did.scope,
+                            'name': did.name,
+                            'did_type': 'Not available with {} plugin.'.format(self.get_plugin_name()),
```

Comment:
I think it would be better to set these to `None`. Just to make sure this isn't accidentally processed otherwise.

Reply:
Yep, no problem.